main.py

```diff
@@ -66,6 +66,8 @@ def setup(self):
         self.link_status_change = defaultdict[str, dict](dict)
         Link.register_status_func(f"{self.napp_id}_link_up_timer",
                                   self.link_status_hook_link_up_timer)
+        Link.register_status_reason_func(f"{self.napp_id}_mismatched_link",
```

Comment:
Nice. 

You gotta also: `Link.register_status_func(f"{self.napp_id}_mismatched_link". self.<TODO>)`

`status_reason` and `status` go hand and hand, so whenever a NApp hook can return a non empty set in a `status_reason`, then it also gotta return `EntityStatus.DOWN` in the respective `status` hook to, and return `None` if it's not mismatched. Here's an example on `of_lldp` too https://github.com/kytos-ng/of_lldp/blob/master/main.py#L48-L51

Without this you can get a situation where status_reason isn't empty but still `status` is UP (which would be incoherent for the logic that we implement and follow):

![20250605_104043](https://github.com/user-attachments/assets/276aa26c-952b-4d95-b702-b88ec14aefc4)

Reply:
Added `status` method.